src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,173 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int currentAmount = 0;
+    private transient Node<T> first;
+    private transient Node<T> last;
+
+    public void check(int index) {
+        if (index < 0 || index >= currentAmount) {
+            throw new ArrayIndexOutOfBoundsException("Non-valid index");
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        int previousAmount = currentAmount;
+        addElement(value);
+        return currentAmount - previousAmount != 0;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != currentAmount) {
+            check(index);
+        }
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == currentAmount) {
+            addLast(value);
+        } else {
+            Node<T> nodeIndex = nodeBy(index);
+            nodeIndex.prev = new Node<>(nodeIndex.prev, value, nodeIndex);
```

Comment:
if you add element to the middle of linkedlist you oldNode.previous.next should point not to the oldNode but to the currently created node, add this logic to method. Because if you iterate node= node.next, you will skip your new node